place_tracker/lib/place_map.dart

```diff
@@ -0,0 +1,437 @@
+              onSavePressed: _confirmAddPlace,
+              onCancelPressed: _cancelAddPlace,
+            ),
+            _AddPlaceFab(
+              visible: _pendingMarker == null,
+              onPressed: _onAddPlaceFabPressed,
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class _CategoryButtonBar extends StatelessWidget {
+  const _CategoryButtonBar({
+    Key key,
+    @required this.selectedPlaceCategory,
+    @required this.visible,
+    @required this.onChanged,
+  }) : assert(selectedPlaceCategory != null),
+       assert(visible != null),
+       assert(onChanged != null),
+       super(key: key);
+
+  final PlaceCategory selectedPlaceCategory;
+  final bool visible;
+  final ValueChanged<PlaceCategory> onChanged;
+
+  @override
+  Widget build(BuildContext context) {
+    return Opacity(
+      opacity: visible ? 1.0 : 0.0,
+      child: Container(
+        padding: const EdgeInsets.fromLTRB(0.0, 0.0, 0.0, 14.0),
+        alignment: Alignment.bottomCenter,
+        child: ButtonBar(
+          alignment: MainAxisAlignment.center,
+          children: <Widget>[
+            RaisedButton(
+              color: selectedPlaceCategory == PlaceCategory.favorite
+                ? Colors.green[700]
+                : Colors.lightGreen,
+              child: const Text(
+                'Favorites',
+                style: TextStyle(color: Colors.white, fontSize: 14.0),
+              ),
+              onPressed: () => onChanged(PlaceCategory.favorite),
+            ),
+            RaisedButton(
+              color: selectedPlaceCategory == PlaceCategory.visited
+                ? Colors.green[700]
+                : Colors.lightGreen,
+              child: const Text(
+                'Visited',
+                style: TextStyle(color: Colors.white, fontSize: 14.0),
+              ),
+              onPressed: () => onChanged(PlaceCategory.visited),
+            ),
+            RaisedButton(
+              color: selectedPlaceCategory == PlaceCategory.wantToGo
+                ? Colors.green[700]
+                : Colors.lightGreen,
+              child: const Text(
+                'Want To Go',
+                style: TextStyle(color: Colors.white, fontSize: 14.0),
+              ),
+              onPressed: () => onChanged(PlaceCategory.wantToGo),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class  _AddPlaceButtonBar extends StatelessWidget {
+  const _AddPlaceButtonBar({
+    Key key,
+    @required this.visible,
+    @required this.onSavePressed,
+    @required this.onCancelPressed,
+  }) : assert(visible != null),
+       assert(onSavePressed != null),
+       assert(onCancelPressed != null),
+       super(key: key);
+
+  final bool visible;
+  final ArgumentCallback<BuildContext> onSavePressed;
+  final VoidCallback onCancelPressed;
+
+  @override
+  Widget build(BuildContext context) {
+    return Builder(builder: (BuildContext context) {
```

Comment:
This Builder doesn't appear to be needed.

Reply:
Removed builder and added one around the scaffold's body. Also added a comment.